lib/DDG/Spice/PublicHolidays.pm

```diff
@@ -0,0 +1,121 @@
+package DDG::Spice::PublicHolidays;
+
+# ABSTRACT: Provides the list of public holidays in a calendar year for a country. 
+
+use strict;
+use DDG::Spice;
+use Locale::Country;
+
+spice is_cached => 1;
+spice proxy_cache_valid => "200 30d";
+spice wrap_jsonp_callback => 0;
+
+spice from => "(.+)/(.+)";
+spice to => 'https://api.xmltime.com/holidays?accesskey={{ENV{DDG_SPICE_TIME_AND_DATE_ACCESSKEY}}}&secretkey={{ENV{DDG_SPICE_TIME_AND_DATE_SECRETKEY}}}&version=2&callback={{callback}}&types=federal&country=$1&year=$2';
+
+my @triggers  = ('public holidays', 'national holidays', 'bank holidays', 'federal holidays');
+my $triggers  = join('|', @triggers);
+my $countries = join('|', share("countries.txt")->slurp(chomp => 1));
+
+triggers any => @triggers;
+    
+handle query_lc => sub {
+    my $query = $_;
+    
+    # Current year and users location are the defaults unless otherwise specified by the query
+    my $defaultYear = (localtime(time))[5] + 1900;
+    my $defaultCountry = $loc->country_name;
+    
+    # Regexes to match components of queries relevant to this IA
+    my $holidays = qr/$triggers/;
+    my $country  = qr/$countries/;
+    my $year     = qr/[1-9]{1}[0-9]{3}/;
+    
+    # Regexes to ignore optional words that can precede the year or country name
+    my $in       = qr/(?:in )?/;
+    my $inThe    = qr/(?:in the |in )?/;
+    
+    my $chosenYear;
+    my $chosenCountry;
+    
+    # Match: <trigger>
```

Comment:
>With the approach you suggested whenever a trigger, country and year appear in the query this IA will fire, regardless of the composition of the query or any extra words which may alter the meaning.

@sekhavati you're correct. If we remove any elements we find from the query and the query still has a length then we could assume it's irrelevant. 

We should at least be able to cut your approach down to fewer compounded regexes by using `|` and `?`:

E.g. `$query =~ /^$holidays $inThe($country)$/` + `/^$holidays $in($year)$/` = `/^$holidays (?:$in($year)|$inThe($country))$/`

> Perhaps we could talk more offline?
 
Not a problem! Send me a message on Slack and we can chat